File: app/src/main/java/org/cerion/stockcharts/charts/ParametersEditControl.java
package org.cerion.stockcharts.charts;

import android.content.Context;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.AttributeSet;
import android.widget.EditText;
import android.widget.LinearLayout;

import org.cerion.stockcharts.R;

import java.util.List;

public class ParametersEditControl extends LinearLayout {

    private Number[] defaultParameters;

    public ParametersEditControl(Context context) {
        super(context);
    }

    public ParametersEditControl(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public void setParameters(final List<Number> params) {

        // TODO add binding so EditText updates corresponding array at all times

        defaultParameters = params.toArray(new Number[0]);
        //final EditText[] fields = new EditText[params.length];

        removeAllViews();
        for(int i = 0; i < params.size(); i++) {
            Number n = params.get(i);
            final int pos = i;
            EditText et = getInputField(n);

            et.addTextChangedListener(new TextWatcher() {
                @Override
                public void beforeTextChanged(CharSequence s, int start, int count, int after) {

                }

                @Override
                public void onTextChanged(CharSequence s, int start, int before, int count) {
                    // TryParse
                    Number defValue = defaultParameters[pos];
                    Number newValue = tryParseNumber(s.toString(), defValue);

                    params.set(pos, newValue);
                }

                @Override
                public void afterTextChanged(Editable s) {

                }
            });
            addView(et);
        }
    }

    public Number[] getParameters() {
        LinearLayout parameters = (LinearLayout)findViewById(R.id.parameters);
        Number p[] = defaultParameters.clone();

        if(p.length != parameters.getChildCount())
            throw new IllegalStateException("expected parameters do not match layout count");

        for (int i = 0; i < p.length; i++) {
            EditText field = (EditText)parameters.getChildAt(i);
            try {
                String entered = field.getText().toString();
                if (p[i] instanceof Integer) {
                    p[i] = Integer.parseInt(entered);
                } else {
                    p[i] = Float.parseFloat(entered);
                }
            } catch(Exception e) {
                // TODO may be unnecessary, we are only getting values
                // Reset input field to default value
                field.setText(p[i].toString());
            }
        }

        return p;
    }

    private Number tryParseNumber(String text, Number defaultVal) {
        try {
            if (defaultVal instanceof Integer) {
                return Integer.parseInt(text);
            } else {
                return Float.parseFloat(text);
            }
        } catch(Exception e) {
            return defaultVal;
        }
    }

    private EditText getInputField(Number n) {
        final EditText input = new EditText(getContext());
        input.setText(n.toString());
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.MATCH_PARENT);
        input.setLayoutParams(lp);

        return input;
    }
}

File: app/src/main/java/org/cerion/stockcharts/ui/charts/views/ParametersEditControl.kt
package org.cerion.stockcharts.ui.charts.views

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.util.AttributeSet
import android.view.View
import android.widget.EditText
import android.widget.LinearLayout
import org.cerion.stockcharts.R

class ParametersEditControl : LinearLayout {
    private var defaultParameters: Array<Number>? = null

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attributeSet: AttributeSet?) : super(context, attributeSet)

    // TODO should be non-mutablelist, this sets the fields then GETS a modified version later
    fun setParameters(params: MutableList<Number>) { // TODO add binding so EditText updates corresponding array at all times
        defaultParameters = params.toTypedArray()
        val currentParameters=  defaultParameters!!

        removeAllViews()

        for (i in params.indices) {
            val n = params[i]
            val et = createInputField(n)

            et.addTextChangedListener(object : TextWatcher {
                override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
                override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) { // TryParse
                    val defValue = currentParameters[i]
                    val newValue = tryParseNumber(s.toString(), defValue)
                    params[i] = newValue
                }

                override fun afterTextChanged(s: Editable) {}
            })
            addView(et)
        }
    }

    val parameters: Array<Number>
        get() {
            val parameters = findViewById<View>(R.id.parameters) as LinearLayout
            val p = defaultParameters!!.clone()
            check(p.size == parameters.childCount) { "expected parameters do not match layout count" }

            for (i in p.indices) {

                val currField = parameters.getChildAt(i) as EditText
                try {
                    val entered = currField.text.toString()
                    if (p[i] is Int)
                        p[i] = entered.toInt()
                    else
                        p[i] = entered.toFloat()

                } catch (e: Exception) {
                    // TODO may be unnecessary, we are only getting values
                    currField.setText(p[i].toString())
                }
            }

            return p
        }

    private fun tryParseNumber(text: String, defaultVal: Number): Number {
        return try {
            when(defaultVal) {
                is Int -> text.toInt()
                is Float -> text.toFloat()
                else -> throw NotImplementedError()
            }
        }
        catch (e: Exception) {
            defaultVal
        }
    }

    private fun createInputField(n: Number): EditText {
        val input = EditText(context)
        input.setText(n.toString())
        val lp = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.MATCH_PARENT)
        input.layoutParams = lp
        return input
    }
}